File: leetcode/lc-11.js
var maxArea = function(height) {
  let l = 0;
  let r = height.length -1 ;
  let res = Math.min(height[l], height[r]) * (r - l);
  while (l < r) {
    if (height[l] < height[r]) {
      l++;
    } else {
      r--;
    }
    res = Math.max(res, Math.min(height[l], height[r]) * (r - l));
  }
  return res;
};

File: leetcode/lc-11.java
public class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        int res = 0;
        while (i < j) {
            int w = j - i;
            int h = Math.min(height[i], height[j]);
            res = Math.max(res, w * h);
            if (height[i] < height[j]) { i++; }
            else { j--; }
        }
        return res;
    }
}
